op-challenger/game/fault/trace/split/provider.go

```diff
@@ -0,0 +1,72 @@
+package split
+
+import (
+	"context"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/types"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var _ types.TraceProvider = (*SplitTraceProvider)(nil)
+
+// SplitTraceProvider is a [types.TraceProvider] implementation that
+// routes requests to the correct internal trace provider based on the
+// depth of the requested trace.
+type SplitTraceProvider struct {
+	logger         log.Logger
+	topProvider    types.TraceProvider
+	bottomProvider types.TraceProvider
```

Comment:
This isn't going to work either - it needs to be a factory for the bottom provider as the trace will differ based on which output roots its going between. I'd forgotten that detail until I re-read the issue too actually. :)

> Should accept a  TraceProvider that provides the "top" of the game and a factory to create a TraceProvider given the the pre and post claims from the top level game. In the OUTPUT_ROOT_CANNON game type those pre and post claims will be the agreed output proposal and the first disputed output proposal.

We could probably do that as a separate PR though.

Reply:
Since we're going to need to work towards this issue in other prs anyway, will put in a new pr so these changes can land